hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactInternalGenericRecord.java

```diff
@@ -187,12 +187,12 @@ public byte getByte(@Nonnull String fieldName) {
         }
     }
 
-    boolean isFieldExists(@Nonnull String fieldName, @Nonnull FieldType type) {
+    boolean isFieldExists(@Nonnull String fieldName, @Nonnull FieldKind type) {
         FieldDescriptor field = schema.getField(fieldName);
         if (field == null) {
             return false;
         }
-        return field.getType().equals(type);
+        return field.getKind().equals(type);
```

Comment:
Minor: Similar to above, I would use `==` and rename the variable from `type` to `kind`